### PR TITLE
wp cli autocomplete v0.3

### PR DESCRIPTION
**Pull request by floq-design**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
**NEW** autocomplete set for `wp-cli`

**What is the current behavior? (You can also link to an open issue here)**
_N/A_

**What is the new behavior (if this is a feature change)?**
_N/A_

**Additional info:**
I've labelled this as a v0.1 release as it's not had much testing - yet!
My intention is to go through and double check all args & options over next few days.
In mean time I thought be good to get it out there for people to feedback on.
Also intend to add some generators for parts like activating/deactivating plugins with dynamic plugin list.

Cheers
James

### CODE REVIEW

**Comment by floq-design** [comment]
### UPDATE
**Release** `v0.2`

- Double-checking of args / options is complete.
- Added descriptions to everything.
- Fixed some typos, added & removed options where appropriate.

Generators to come if I find the time - invested over 20 hours in this already!

Enjoyed putting it together but maybe time for someone else to inject some :heart:

Cheers
James

File: dev/wp.ts [hunk 1]
```diff
@@ -0,0 +1,12335 @@
+// To learn more about Fig"s autocomplete standard visit: https://fig.io/docs/autocomplete/building-a-spec#building-your-first-autocomplete-spec
+const global_parameter_path = {
+  name: "--path",
+  insertValue: "--path=",
+  description: "Path to the WordPress files.",
+  args: {
+    name: "path",
+  },
```
Comment (QuiiBz) [inline dev/wp.ts:8]: I think this can have a template of `filepaths`

File: dev/wp.ts [hunk 2]
```diff
@@ -0,0 +1,12335 @@
...
+    //   // If a subcommand or option takes an argument, you must include the args prop, even if it"s an empty object (like below)
+    //   // If you want to build custom suggestions for arguments check out: https://fig.io/docs/autocomplete/building-a-spec#making-advanced-suggestions
+    //   args: [{}],
+    //   options: global_parameters,
+    // },
+    {
+      name: "cache",
+      description:
+        "Adds, removes, fetches, and flushes the WP Object Cache object.",
+      args: [{}],
```
Comment (QuiiBz) [inline dev/wp.ts:183]: You can just set an empty object here. But does `wp cache` takes any arguments?

File: dev/wp.ts [hunk 3]
```diff
@@ -0,0 +1,12335 @@
...
+          name: "type",
+          description:
+            "Attempts to determine which object cache is being used.",
+        },
+      ],
+    },
+    {
+      name: "cap",
+      description: "Adds, removes, and lists capabilities of a user role.",
+      args: [{}],
```
Comment (QuiiBz) [inline dev/wp.ts:362]: Same as above

File: dev/wp.ts [hunk 4]
```diff
@@ -0,0 +1,12335 @@
...
+                  name: "--set-url",
+                  insertValue: "--set-url=",
+                  description: "Set url for alias.",
+                  args: { name: "url" },
+                },
+                {
+                  name: "--set-path",
+                  insertValue: "--set-path=",
+                  description: "Set path for alias.",
+                  args: { name: "path" },
```
Comment (QuiiBz) [inline dev/wp.ts:517]: Could this arg have a template of `filepaths`?

File: dev/wp.ts [hunk 5]
```diff
@@ -0,0 +1,12335 @@
...
+                "Download directly from a provided URL instead of fetching the URL from the wordpress.org servers.",
+            },
+          ],
+          options: [
+            {
+              name: "--path",
+              insertValue: "--path=",
+              description:
+                "Specify the path in which to install WordPress. Defaults to current directory.",
+              args: { name: "path" },
```
Comment (QuiiBz) [inline dev/wp.ts:1834]: Could this arg have a template of `filepaths`?

File: dev/wp.ts [hunk 6]
```diff
@@ -0,0 +1,12335 @@
...
+    {
+      name: "export",
+      description: "Exports WordPress content to a WXR file.",
+      options: [
+        {
+          name: "--dir",
+          insertValue: "--dir=",
+          description:
+            "Full path to directory where WXR export files should be stored. Defaults to current working directory.",
+          args: { name: "dirname" },
```
Comment (QuiiBz) [inline dev/wp.ts:3384]: Could this arg have a template of `folders`?

**Review by QuiiBz — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 7]
```diff
@@ -14,5 +14,6 @@ module.exports = {
     "no-unused-vars": ["off"],
     "no-var": ["off"],
     "@typescript-eslint/no-unused-vars": ["off"],
+    "quotes": ["error", "double"],
```
[the remark this answers is not in the document]
Comment (floq-design) [reply .eslintrc.js]: I added it to over-ride my global setting.. I've reverted it in latest commit

File: dev/wp.ts [hunk 8]
```diff
@@ -0,0 +1,12335 @@
+// To learn more about Fig"s autocomplete standard visit: https://fig.io/docs/autocomplete/building-a-spec#building-your-first-autocomplete-spec
+const global_parameter_path = {
+  name: "--path",
```
[the remark this answers is not in the document]
Comment (floq-design) [reply dev/wp.ts]: 👍

File: dev/wp.ts (hunk 1 above)
Comment (floq-design) [reply dev/wp.ts]: 👍

File: dev/wp.ts (hunk 2 above)
Comment (floq-design) [reply dev/wp.ts:184]: Tried changing to `args: {},` but got error for following `option: ...` & `suggestions: ...` so left as is.

`wp cache` has no arguments - only options & subcommands.
https://developer.wordpress.org/cli/commands/cache/

File: dev/wp.ts (hunk 3 above)
Comment (floq-design) [reply dev/wp.ts:363]: Ditto - same as `wp cache`

File: dev/wp.ts (hunk 4 above)
Comment (floq-design) [reply dev/wp.ts]: 👍

File: dev/wp.ts (hunk 5 above)
Comment (floq-design) [reply dev/wp.ts]: 👍

File: dev/wp.ts (hunk 6 above)
Comment (floq-design) [reply dev/wp.ts]: 👍

File: dev/wp.ts (hunk 2 above)
Comment (QuiiBz) [reply dev/wp.ts:184]: Alright, then you can remove it completely

Comment (floq-design) [reply dev/wp.ts]: Removed all empty `args`

**Review by QuiiBz — APPROVED** [review]
Thank you so much for this contribution!

**Comment by QuiiBz** [comment]
Merged 🚀 thanks a lot for this one!

**Pull request merged**